datafusion/groupby-datafusion.py

```diff
@@ -47,163 +47,173 @@ def ans_shape(batches):
 
 question = "sum v1 by id1" # q1
 gc.collect()
-t_start = timeit.default_timer()
-ans = ctx.sql("SELECT id1, SUM(v1) AS v1 FROM x GROUP BY id1").collect()
-shape = ans_shape(ans)
-print(shape, flush=True)
-t = timeit.default_timer() - t_start
-m = memory_usage()
-t_start = timeit.default_timer()
-df = ctx.create_dataframe([ans])
-chk = df.aggregate([], [f.sum(col("v1"))]).collect()[0].column(0)[0]
-chkt = timeit.default_timer() - t_start
-write_log(task=task, data=data_name, in_rows=in_rows, question=question, out_rows=shape[0], out_cols=shape[1], solution=solution, version=ver, git=git, fun=fun, run=1, time_sec=t, mem_gb=m, cache=cache, chk=make_chk([chk]), chk_time_sec=chkt, on_disk=on_disk)
-del ans
-gc.collect()
+for run in range(1, 2):
```

Comment:
i am not maintainer of this repo, so it is not up to me anymore to request any changes about that, but... we used to keep it simple and reproducible line by line. I understand well reasons for tidying up the code and putting it into 1-2 loop, but then code is not simple to run line by line AND we could possibly do same tidying up for other solutions, and for each programming language it will be different, or in some cases impossible (clickhouse). Therefore the most universal approach we have taken was to keep it simply to be able to run line by line from the terminal connected to remote server. It actually was very useful in debugging.

Reply:
Ok, I keep the decision to @Tmonster . I could revert the change and copy the code :)